code/game/turfs/simulated/floor.dm

```diff
@@ -87,6 +88,9 @@
 
 /turf/open/floor/proc/update_icon()
 	update_visuals()
+	cut_overlays()
```

Comment:
`update_visuals()` is responsible for atmos overlays, like the icy stuff we added before. Calling `cut_overlays()` right after will remove that. No bueno.

Will be best if you do `cut_overlay(current_overlay)` instead, though I'm actually not sure if that will work. Do try!